epochStart/bootstrap/process_test.go

```diff
@@ -575,74 +1102,449 @@ func TestSyncUserAccountsState(t *testing.T) {
+		coreComp.IntMarsh = &testscommon.MarshalizerStub{
+			MarshalCalled: func(obj interface{}) ([]byte, error) {
+				return nil, expectedErr
+			},
+		}
+
+		args := createMockEpochStartBootstrapArgs(coreComp, cryptoComp)
+
+		prevShardHeader := &block.Header{}
+		notarizedShardHeader := &block.Header{
+			PrevHash: prevShardHeaderHash,
+		}
+		notarizedMetaHeader := &block.MetaBlock{
+			PrevHash: prevMetaHeaderHash,
+		}
+		metaBlock := &block.MetaBlock{
+			Epoch: 2,
+			EpochStart: block.EpochStart{
+				LastFinalizedHeaders: []block.EpochStartShardData{
+					{
+						HeaderHash:            notarizedShardHeaderHash,
+						ShardID:               0,
+						FirstPendingMetaBlock: notarizedMetaHeaderHash,
+					},
+				},
+			},
+		}
+		prevMetaBlock := &block.MetaBlock{
+			Epoch: 1,
+			EpochStart: block.EpochStart{
+				LastFinalizedHeaders: []block.EpochStartShardData{
+					{
+						HeaderHash: notarizedShardHeaderHash,
+						ShardID:    0,
+					},
+				},
+			},
+		}
+
+		epochStartProvider, _ := NewEpochStartBootstrap(args)
+		epochStartProvider.syncedHeaders = make(map[string]data.HeaderHandler)
+		epochStartProvider.epochStartMeta = metaBlock
+		epochStartProvider.prevEpochStartMeta = prevMetaBlock
+		epochStartProvider.headersSyncer = &epochStartMocks.HeadersByHashSyncerStub{
+			GetHeadersCalled: func() (m map[string]data.HeaderHandler, err error) {
+				return map[string]data.HeaderHandler{
+					string(notarizedShardHeaderHash): notarizedShardHeader,
+					string(notarizedMetaHeaderHash):  notarizedMetaHeader,
+					string(prevShardHeaderHash):      prevShardHeader,
+				}, nil
+			},
+		}
+		epochStartProvider.dataPool = &dataRetrieverMock.PoolsHolderStub{
+			TrieNodesCalled: func() storage.Cacher {
+				return &testscommon.CacherStub{
+					GetCalled: func(key []byte) (value interface{}, ok bool) {
+						return nil, true
+					},
+				}
+			},
+		}
+
+		epochStartProvider.miniBlocksSyncer = &epochStartMocks.PendingMiniBlockSyncHandlerStub{}
+		epochStartProvider.requestHandler = &testscommon.RequestHandlerStub{}
+		epochStartProvider.nodesConfig = &sharding.NodesCoordinatorRegistry{}
+
+		err := epochStartProvider.requestAndProcessForShard()
+		assert.Equal(t, expectedErr, err)
+	})
+}
+
+func TestRequestAndProcessForMeta_ShouldFail(t *testing.T) {
+	coreComp, cryptoComp := createComponentsForEpochStart()
+
+	notarizedShardHeaderHash := []byte("notarizedShardHeaderHash")
+	prevShardHeaderHash := []byte("prevShardHeaderHash")
+
+	t.Run("fail to create storage handler component", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockEpochStartBootstrapArgs(coreComp, cryptoComp)
+
+		metaBlock := &block.MetaBlock{
+			Epoch: 2,
+			EpochStart: block.EpochStart{
+				LastFinalizedHeaders: []block.EpochStartShardData{
+					{HeaderHash: notarizedShardHeaderHash, ShardID: 0},
+				},
+			},
+		}
+
+		epochStartProvider, _ := NewEpochStartBootstrap(args)
+		epochStartProvider.syncedHeaders = make(map[string]data.HeaderHandler)
```

Comment:
```suggestion
```

Again, a bit "fishy" that these funcs rely on the certain fact that `epochStartMeta` is not nil and you have to initialize it.

Reply:
agree :+1:

Reply:
In this case, it could be removed